datafusion/core/src/config.rs

```diff
@@ -237,6 +237,15 @@ impl BuiltInConfigs {
                  to reduce the number of rows decoded.",
                 false,
             ),
+            #[cfg(test)]
```

Comment:
I think this `#[cfg(test)]` will effectively only apply to tests in the `datafusion/core` crate -- this option will be false for other tests (like `sql_integration` for example)